### PR TITLE
DO NOT MERGE! This is for review to be added to #201

### PR DESCRIPTION
**Pull request by rwaldron**
Once review here is complete, I will cherry pick the commit to the branch at https://github.com/salesforce/near-membrane/pull/201

### CODE REVIEW

File: packages/near-membrane-base/src/__tests__/environment.spec.ts [hunk 1]
```diff
@@ -454,7 +472,7 @@ describe('VirtualEnvironment', () => {
             expect(count).toBe(0);
         });
 
-        it('calls a lazy endowment setter, called after remap', () => {
+        xit('calls a lazy endowment setter, called after remap', () => {
```
Comment (rwaldron) [inline packages/near-membrane-base/src/__tests__/environment.spec.ts:475]: There are four tests that may be invalid. I've "x"'ed them for now.

File: packages/near-membrane-base/src/__tests__/environment.spec.ts [hunk 2]
```diff
@@ -1,5 +1,5 @@
 import { SupportFlagsField } from '../../types';
-import { getFilteredEndowmentDescriptors, VirtualEnvironment } from '../index';
+import { getFilteredGlobalObjectShapeDescriptors, VirtualEnvironment } from '../index';
```
Comment (jdalton) [inline packages/near-membrane-base/src/__tests__/environment.spec.ts:2]: ☝️ That name is so nice!

File: packages/near-membrane-base/src/__tests__/environment.spec.ts [hunk 3]
```diff
@@ -326,14 +332,17 @@ describe('VirtualEnvironment', () => {
...
-            ve.lazyRemap(redValue, Object.keys(getFilteredEndowmentDescriptors(endowments)));
+            ve.lazyRemap(
+                redValue,
+                Object.keys(getFilteredGlobalObjectShapeDescriptors(endowments))
+            );
             expect(redValue.d).toBe(0);
         });
 
-        it('will not call an endowment getter that does not exist', () => {
+        xit('will not call an endowment getter that does not exist', () => {
```
Comment (jdalton) [inline packages/near-membrane-base/src/__tests__/environment.spec.ts:345]: ☝️ why skip?

File: packages/near-membrane-base/src/__tests__/intrinsics.spec.ts [hunk 4]
```diff
@@ -1,103 +1,29 @@
...
-        // Ignoring "Property 'assertions' does not exist on type '{...}'."
-        // @ts-ignore
-        expect.assertions(remappedIntrinsicNames.length);
-        remappedIntrinsicNames.forEach((remappedIntrinsicName) => {
-            expect(isIntrinsicGlobalName(remappedIntrinsicName)).toBe(false);
-        });
-    });
-    it('should return true for all non-remapped ES global names', () => {
+describe('getFilteredGlobalObjectShapeDescriptors()', () => {
+    it('ignores ES built-ins', () => {
```
Comment (jdalton) [inline packages/near-membrane-base/src/__tests__/intrinsics.spec.ts:4]:
```suggestion
    it('ignores non-remapped ES intrinsics', () => {
```

File: packages/near-membrane-base/src/__tests__/intrinsics.spec.ts [hunk 5]
```diff
@@ -1,103 +1,29 @@
...
+        const filteredEndowmentDescriptors = getFilteredGlobalObjectShapeDescriptors({
+            Math,
         });
+        expect(filteredEndowmentDescriptors.Math).toBe(undefined);
     });
-});
-
-describe('getFilteredEndowmentDescriptors()', () => {
-    it('ignores ES built-ins', () => {
+    it('includes remapped ES built-ins', () => {
```
Comment (jdalton) [inline packages/near-membrane-base/src/__tests__/intrinsics.spec.ts:13]:
```suggestion
    it('includes remapped ES intrinsics', () => {
```

File: packages/near-membrane-base/src/intrinsics.ts [hunk 6]
```diff
@@ -27,12 +28,13 @@ const {
  * can be considered equivalents (without identity discontinuity).
  */
 const ESGlobalKeys = new Set([
-    // *** 18.1 Value Properties of the Global Object
+    // *** 19.1 Value Properties of the Global Object
+    'globalThis',
```
Comment (jdalton) [inline packages/near-membrane-base/src/intrinsics.ts:32]: Should we export these https://github.com/salesforce/near-membrane/blob/master/packages/near-membrane-base/src/intrinsics.ts#L29 and use them here (I noticed `globalThis` is not part of the list there or maybe not the first and missed it there)

File: packages/near-membrane-base/src/intrinsics.ts [hunk 7]
```diff
@@ -147,13 +154,19 @@ export function getFilteredEndowmentDescriptors(endowments: object): PropertyDes
...
-export function isIntrinsicGlobalName(key: PropertyKey): boolean {
-    return ReflectApply(SetProtoHas, ESGlobalKeys, [key]);
+export function getResolvedShapeDescriptors(
+    globalObjectShape: typeof globalThis,
+    endowments: object
+): PropertyDescriptorMap {
+    const globalObjectShapeDescriptors = getFilteredGlobalObjectShapeDescriptors(globalObjectShape);
+    const endowmentsDescriptors = getFilteredGlobalObjectShapeDescriptors(endowments);
+    // eslint-disable-next-line prefer-object-spread
+    return ObjectAssign({}, globalObjectShapeDescriptors, endowmentsDescriptors);
```
Comment (jdalton) [inline packages/near-membrane-base/src/intrinsics.ts:171]: Null proto for the returned object here? I noticed it is nulled in `getFilteredGlobalObjectShapeDescriptors`

**Comment by jdalton** [comment]
Overall good! I have a few minor optimizations I'll do in a follow-up PR.

File: packages/near-membrane-base/src/intrinsics.ts (hunk 7 above)
Comment (rwaldron) [reply packages/near-membrane-base/src/intrinsics.ts:171]: Oh, good call.

Comment (rwaldron) [reply packages/near-membrane-base/src/intrinsics.ts:171]:
```suggestion
    return ObjectAssign({ __proto__: null }, globalObjectShapeDescriptors, endowmentsDescriptors);
```

**Pull request closed without merging**